src/tools/ImageTransformer.cpp

```diff
@@ -0,0 +1,27 @@
+// Created by Maxandre Ogeret.
+// (c) 2022 University of Tartu - Autonomous Driving Lab.
+
+#include "tools/ImageTransformer.h"
+
+ImageTransformer::ImageTransformer(DriveworksApiWrapper* driveworksApiWrapper)
+  : driveworksApiWrapper_(driveworksApiWrapper)
+{
+  CHK_DW(dwImage_create(&imgRgb8_, camera_common::get_rgba_pitch_img_prop(), driveworksApiWrapper_->context_handle_));
+
+  dwImageTransformation_initialize(&ImageTransformationEngine_, {}, driveworksApiWrapper_->context_handle_);
+  dwImageTransformation_setInterpolationMode(DW_IMAGEPROCESSING_INTERPOLATION_DEFAULT, ImageTransformationEngine_);
+}
+
+ImageTransformer::~ImageTransformer()
+{
+  dwImageTransformation_release(ImageTransformationEngine_);
+  dwImage_destroy(imgRgb8_);
+}
+
+void ImageTransformer::transform_image(dwImageHandle_t* input, dwImageHandle_t* output)
+{
+  CHK_DW(dwImage_copyConvert(imgRgb8_, *input, driveworksApiWrapper_->context_handle_));
+  ROS_INFO_STREAM_ONCE("Transformer convert OK");
```

Comment:
Is it necessary to have those log messages in this tight loop? Even when they output the messages only once, there is still some work done to check if the message is duplicate, whether severity level exceeds threshold etc.

Reply:
I dont think it has a lot of impact but I also haven't measured it.
I removed the log because they were not that useful.